From 186c4b0fa66ce6e829548ebbcda0f8f0a8e1e882 Mon Sep 17 00:00:00 2001
From: Timothy Brown <timothy.brown-1@colorado.edu>
Date: Mon, 6 Oct 2014 13:11:15 -0600
Subject: [PATCH] Updating the PMI check.

Updated the opal_check_pmi M4 to actually look for the PMI
dynamic library in the provided directory.
---
 config/opal_check_pmi.m4 | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/config/opal_check_pmi.m4 b/config/opal_check_pmi.m4
index e71202c..ec407b3 100644
--- a/config/opal_check_pmi.m4
+++ b/config/opal_check_pmi.m4
@@ -71,9 +71,10 @@ AC_DEFUN([OPAL_CHECK_PMI],[
            # support appends "include" to the path, which won't
            # work with slurm :-(
            AS_IF([test ! -z "$with_pmi" -a "$with_pmi" != "yes"],
-                 [AS_IF([test -d "$with_pmi/lib64"],
+                 [AS_IF([test -f "$with_pmi/lib64/libpmi.so"],
                         [opal_check_pmi_$1_LDFLAGS="-L$with_pmi/lib64"
                          opal_pmi_rpath="$with_pmi/lib64"],
+                        [test -f "$with_pmi/lib/libpmi.so"],
                         [opal_check_pmi_$1_LDFLAGS="-L$with_pmi/lib"
                          opal_pmi_rpath="$with_pmi/lib"])
                   # look for required headers - both pmi.h AND/OR pmi2.h
-- 
1.7.11.1

